### DISCUSSION TITLE
Blazor Project Tracker sample - Best practice for loading required list for display

### DISCUSSION THREAD

**Discussion by Conway6288** [MarimerLLC/csla#4224] [Questions]
The latest Project Tracker sample fetches an instance of RoleList on the [EditProject.razor](https://github.com/MarimerLLC/csla/blob/main/Samples/ProjectTracker/ProjectTracker.Blazor/ProjectTracker.Blazor.Client/Pages/EditProject.razor) page. That page then has a select input bound to the role property on a [ProjectResourceEdit](https://github.com/MarimerLLC/csla/blob/main/Samples/ProjectTracker/ProjectTracker.BusinessLibrary/ProjectResourceEdit.cs) instance. The options for the select are populated from the RoleList instance directly on the EditProject page. I notice that the RoleList object on ProjectResourceEdit which is used for getting the "RoleName" property.

What is the best practice for these list scenarios? The ProjectResourceEdit is the only thing on the page that requires the RoleList by the looks of it so that leaves me with questions:

- Is there any purpose of having the RoleList on the ProjectResourceEdit other than getting the RoleName?
- The list object seems to be used for display purposes only and so should it be on the business object at all if it is really for display only? 
- If it is on the business object, why have it on the EditProject.razor page and not just use the one instantiated from the business object? What is the advantage here.

**Comment by Chicagoan2016** [comment]
@rockfordlhotka  has some "Using Csla" ebooks. I recommend reading them, although they were for version 4, I still find them relevant.
If this is your first time looking into Csla, it would help to understand Csla's object stereotypes and their purpose.

**Reply by Conway6288** [reply]
Thanks for the reply @Chicagoan2016.

I have read the ebooks and have been using CSLA for a good while, however I am diving deeper into using Blazor and was just looking at the Project Tracker Blazor sample. I couldn't understand why you would have an instance of the RoleList on both the EditProjects page and the ProjectResourceEdit object. It seems the only reason to have that list instance on ProjectResourceEdit directly is to get the RoleName. 

I know lots has changed with the latest version of CSLA, just want to follow the best practices for this scenario of displaying the list on screen.
